packages/react/menu/src/Menu.tsx

```diff
@@ -74,13 +73,26 @@ type MenuOwnProps = {
 const Menu: React.FC<MenuOwnProps> = (props) => {
   const { open = false, children, onOpenChange } = props;
   const [content, setContent] = React.useState<MenuContentElement | null>(null);
+  const isUsingKeyboardRef = React.useRef(false);
   const handleOpenChange = useCallbackRef(onOpenChange);
   const computedDirection = useDirection(content, props.dir);
 
+  React.useEffect(() => {
+    const handleKeyDown = () => (isUsingKeyboardRef.current = true);
+    const handlePointerMove = () => (isUsingKeyboardRef.current = false);
+    document.addEventListener('keydown', handleKeyDown, { capture: true });
+    document.addEventListener('pointermove', handlePointerMove, { capture: true });
```

Comment:
I think we should use `pointerdown` no? because on touch screen `pointermove` may not happen, but a `pointerdown` would always happen, or potentially we do both to cover all basis.

Reply:
we would need both, which just got me thinking - I dunno if we need `pointer` events at all because how would you navigate by keyboard on a touch device?

Reply:
You definitely can, mixed input right? iPads have both touch and keyboard support for example.

Reply:
Life is to complicated now, can't assume anything anymore 😀 hahaha

Reply:
haha true, I'll add both events in that case